test/gtest/uct/test_md.cc

```diff
@@ -325,7 +325,9 @@ UCS_TEST_P(test_md, mem_query) {
                                                mem_buf.size(), &mem_attr);
         ASSERT_UCS_OK(status);
         EXPECT_EQ(mem_type, mem_attr.mem_type);
-        EXPECT_NE(UCS_SYS_DEVICE_ID_UNKNOWN, mem_attr.sys_dev);
+        if (mem_buffer::is_device_detected(mem_attr.mem_type)) {
```

Comment:
better do this in the test itself and not in mem_buffer class

Reply:
ok, moved into test_md class